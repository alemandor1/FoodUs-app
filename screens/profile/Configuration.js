import React, { useState, useRef, useEffect } from 'react'
import { StyleSheet, Text, View } from 'react-native'
import { Button, ListItem, Icon } from 'react-native-elements';
import { useNavigation } from '@react-navigation/native'
import { map } from 'lodash';

import { closeSession, getCurrentUser } from '../../utils/actions';
import InfoUser from '../../components/account/InfoUser';
import Profile from '../../screens/profile/Profile'

export default function Configuration() {

    const [loading, setLoading] = useState(false)
    const [loadingText, setLoadingText] = useState("")
    const [user, setUser] = useState(null)

    useEffect(() => {
        setUser(getCurrentUser())
    }, [])

    const navigation = useNavigation()

    const generateOptions = () => {
        return [
            {
                title: "User settings",
                iconNameLeft: "settings",
                iconColorLeft: "#a7bfd3",
                iconNameRight: "chevron-right",
                iconColorRight: "#a7bfd3",
                onPress: () => navigation.navigate('Profile')
            },
            {
                title: "Food list",
                iconNameLeft: "kitchen", 
                iconColorLeft: "#a7bfd3",
                iconNameRight: "chevron-right",
                iconColorRight: "#a7bfd3",
                onPress: () => navigation.navigate('Profile')
            },
            {
                title: "Recipe history",
                iconNameLeft: "history", 
                iconColorLeft: "#a7bfd3",
                iconNameRight: "chevron-right",
                iconColorRight: "#a7bfd3",
                onPress: () => navigation.navigate('HistoryList')
            },
            {
                title: "Favorite recipes",
                iconNameLeft: "favorite-border", 
                iconColorLeft: "#a7bfd3",
                iconNameRight: "chevron-right",
                iconColorRight: "#a7bfd3",
                onPress: () => navigation.navigate('Profile')
            },
            {
                title: "Shopping list",
                iconNameLeft: "shopping-cart", 
                iconColorLeft: "#a7bfd3",
                iconNameRight: "chevron-right",
                iconColorRight: "#a7bfd3",
                onPress: () => navigation.navigate('Profile')
            }
        ]
    }

    const menuOptions = generateOptions();

    return (
        <View style={styles.container}>
            {
                user && (
                    <View>
                        <InfoUser
                            user={user}
                            setLoading={setLoading}
                            setLoadingText={setLoadingText}
                        />
                        <View>
            {
                map(menuOptions, (menu, index) => (
                    <ListItem
                        key={index}
                        style={styles.menuItem}
                        onPress={menu.onPress}
                    >
                        <Icon
                            type="material-comunity"
                            name={menu.iconNameLeft}
                            color={menu.iconColorLeft}
                        />
                        <ListItem.Content>
                            <ListItem.Title>{menu.title}</ListItem.Title>
                        </ListItem.Content>
                        <Icon
                            type="material-comunity"
                            name={menu.iconNameRight}
                            color={menu.iconColorRight}
                        />
                    </ListItem>
                ))
            }
        </View>
                    </View>
                    )
            }
            <Button
                title="Log out"
                buttonStyle={styles.btnCloseSession}
                titleStyle={styles.btnCloseSessionTitle}
                onPress={() => {
                    closeSession()
                    navigation.navigate("Home")
                }}
            />
        </View>
    )
}

const styles = StyleSheet.create({
    container : {
        minHeight: "100%",
        backgroundColor: "#f9f9f9"
    },
    btnCloseSession : {
        marginTop: 30,
        borderRadius: 5,
        backgroundColor: "#F94F62",
        borderTopWidth: 1,
        borderTopColor: "#F6051F",
        borderBottomWidth: 1,
        borderBottomColor: "#F6051F",
        paddingVertical: 10,
        width: "50%",
        alignSelf: "center"
    },
    btnCloseSessionTitle : {
        color: "#FFFFFF"
    },
    menuItem: {
        borderBottomWidth: 1,
        borderBottomColor: "#a7bfd3"
    }
})