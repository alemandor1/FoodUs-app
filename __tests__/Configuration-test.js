import React from "react";
import Configuration from "../screens/profile/Configuration";
import {create} from 'react-test-renderer'
import {render, waitFor} from "@testing-library/react-native";

const tree = create(<Configuration />)

test('snapshot', () => {
    expect(tree).toMatchSnapshot();
}) 

describe("<Configuration />", () => {
    beforeEach(() => {
        component = render(<Configuration />);
    });

    it("Renderiza correctamente", () => {
        expect(component).toBeDefined();
        expect(component.getByTestId("buttonLogOut")).toBeDefined();
        expect(component.queryAllByTestId("list").length).toEqual(1);
    });

    it("Renderiza los elementos despues de llamar a la api", () => {
        waitFor(() => expect(component.getByTestId("list")).toBeDefined());
    })
})