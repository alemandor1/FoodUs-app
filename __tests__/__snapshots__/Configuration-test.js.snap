// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "minHeight": "100%",
    }
  }
  testID="list"
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="buttonLogOut"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#F94F62",
            "borderBottomColor": "#F6051F",
            "borderBottomWidth": 1,
            "borderColor": "#2089dc",
            "borderRadius": 5,
            "borderTopColor": "#F6051F",
            "borderTopWidth": 1,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 30,
            "padding": 8,
            "paddingVertical": 10,
            "width": "50%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Log out
        </Text>
      </View>
    </View>
  </View>
</View>
`;
